conan/cli/commands/install.py

```diff
@@ -39,6 +39,8 @@ def install(conan_api, parser, *args):
                              "the provided patterns")
     parser.add_argument("--build-require", action='store_true', default=False,
                         help='Whether the provided path is a build-require')
+    parser.add_argument("--generate-virtualenvs", default="auto", choices=["auto", "never", "NotEmpty"],
```

Comment:
- Lets try to have a shorter name, maybe ``--generate-envs``? Or some nice shortcut?
- Drop ``NotEmpty`` if not used at this moment. If used, lets try to make it same casing ``not_empty`` or some alternative synonym
- Is it possible to allow the "non-definition" to have a ``None`` value? So ``--generate-virtualenvs=auto`` shouldn't be possible, it shouldn't be even written because it is already the default, totally meaningless to add that arg?